08_traits_and_reading_bytes.md

```diff
@@ -28,46 +46,63 @@ fn main() {
     let version = u32::from_le_bytes(buffer);
 
     println!("Version: {}", version);
+    println!("Bytes slice: {:?}", bytes_slice);
 }
 ```
 
-The `mut` keyword before `bytes_slice` tells Rust the variable is mutable. If we don't provide that keyword in a variable declaration, then the compiler will complain that we're attempting to change an immutable variable, which is not allowed. 
+The `mut` keyword before `bytes_slice` tells Rust the variable is mutable.
+If we don't provide that keyword in a variable declaration, then the compiler will complain that we're attempting to change the value of an immutable variable, which is not allowed.
 
-You might also notice the `&mut` keyword in the argument to the `read` method. This indicates that we're passing in `buffer` as a *mutable reference*. We'll talk more about this means in the next chapter so for now let's not worry about that nuance. 
+You might also notice the `&mut` keyword in the argument to the `read` method.
+This indicates that we're passing in `buffer` as a *mutable reference*.
+We'll talk more about this means in the next chapter so for now let's not worry about that nuance. 
 
 When we run this, it will print the following:
 ```console
 Version: 1
 Bytes slice: [2]
 ```
 
-And this is what we'd expect. The Version is `1`. And the `bytes_slice` variable has been updated and no longer contains the first 4 bytes. 
+And this is what we'd expect.
+The Version is `1` and the `bytes_slice` variable has been updated and no longer contains the first 4 bytes. 
 
-You may notice that the way this works is that you have to first create an array with a fixed size. Calling `read` will then extract the number of bytes equal to the size of the array, store that into a buffer and then update our slice.
+You may notice that the way this works is that you have to first create an array with a fixed size.
+Calling `read` will then extract the number of bytes equal to the size of the array, store that into a buffer and then update our slice.
 
-## Traits Explanation
+## What are traits?
 
-So what are traits exactly? Traits are a way to define shared behavior. You can think of them as a template for a particular set of behaviors. For example, the `Read` trait provides a template for types that want to "read data". It lays out what to expect and what types of functions are available.
+Traits are a way to define shared behavior.
```

Comment:
I don't quite like the adjective `shared` behavior. I personally prefer to call it `abstract` behavior, but it will probably confuse people...

Reply:
Hm I don't mind the word `abstract` since it's followed by the concept of a `template`.

Reply:
Looks like the Rust book uses that language though: https://doc.rust-lang.org/book/ch10-02-traits.html

Reply:
So, let's leave as it is.